src/fetcher.js

```diff
@@ -911,10 +891,11 @@ class Fetcher {
    * @param options {Object}
    * @param errorMessage {string}
    * @param statusCode {number}
+   * @param response {Response}  // when an fetch() error
    *
    * @returns {Promise<Object>}
    */
-  failFetch (options, errorMessage, statusCode) {
+  failFetch (options, errorMessage, statusCode, response) {
```

Comment:
Is it worth making sure that {RDFXML,XHTML,XML,N3}Handler pass the response? Could add it to their signatures and to ['handleResponse`](https://github.com/linkeddata/rdflib.js/blob/fetcher-response/src/fetcher.js#L1485). This expands the API but has no performance impact. Also, I think it could be marked optional in the API because it's optional in failFetch.

Reply:
It would mean that every parser would have to have a copy of the response to it could call it on error?  Or we would wrap each one with a  try/catch which would then add that  information?